content/guides/accessibility/color-considerations.mdx

```diff
@@ -0,0 +1,332 @@
+
+The majority of GitHub input text size is "normal text". Both "normal text" and "large text" are WCAG-specific terms. More information about this can be found on [the Key Terms subsection of SC 1.4.3: Contrast (Minimum)](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html#dfn-large-scale).
+
+#### Button icon contrast
+
+Icons can either be:
+
+1. Functional, or
+2. Decorative.
+
+##### Functional icons
+
+If the icon is the sole means of determining a button's purpose, it is considered functional. An example of this is an [IconButton](https://primer.style/design/components/icon-button).
+
+<img src={ButtonFunctionalIcon} alt="A wireframe illustration of a button. It only has the GitHub Octocat logo in its center, and no text. A label pointing to the icon reads, 'Communicates purpose.' A label pointing to the button's background and border reads, 'Reinforces interactive area.'" />
+
+Functional icons **must** meet a contrast ratio of `3:1`. This is calculated by the icon's computed color value compared to the computed background color the input is placed on.
+
+Beyond the scope of normative WCAG guidance, a further consideration is if the icon's meaning has cultural considerations or represents abstract functionality a user may not be familiar with.
+
+##### Decorative icons
+
+Buttons with a text label that **also** have an accompanying icons can consider their icons as decorative. Decorative icons **do not** need to meet WCAG-compliant contrast ratios.
+
+The reason for this is that the text label communicates the input's purpose, provided its computed color value is WCAG compliant. The icon in this circumstance reinforces the text label. 
+
+An example of this is a button with both a [trash Octicon](https://primer.style/design/foundations/icons/trash-16) **and** a text label of "Delete".
+
+<img src={ButtonDecorativeIcon} alt="A wireframe illustration of a button. It only has the GitHub Octocat logo in its center, and no text. A label pointing to the icon reads, 'Communicates purpose.' A label pointing to the button's background and border reads, 'Reinforces interactive area.'" />
+
+Beyond the scope of normative WCAG guidance, a further consideration is if making a decorative icon color contrast compliant is useful for aiding in facilitating understanding, even if it is not required to be.
+
+#### Button border contrast
+
+The contrast of a button's border color compared to the input's background color **does not** need to meet a contrast ratio. Only the [button's text](#button-text-contrast) or [functional icon](#functional-icons) does.
+
+#### Button disabled state contrast
+
+Broadly-speaking, buttons **should not** be disabled until form validation criteria has been met. An example of behaviro to avoid is disabling a submit button until a user has filled out all required fields.
```

Comment:
```suggestion
Broadly-speaking, buttons **should not** be disabled until form validation criteria has been met. An example of behavior to avoid is disabling a submit button until a user has filled out all required fields.
```